src/helics/common/JsonProcessingFunctions.cpp

```diff
@@ -16,15 +16,34 @@ SPDX-License-Identifier: BSD-3-Clause
 
 namespace helics::fileops {
 
-bool looksLikeCommandLine(std::string_view testString)
+bool looksLikeJson(std::string_view jsonString)
 {
-    if (testString.empty()) {
+    if (jsonString.empty()) {
         return false;
     }
-    if (testString.front() == '-') {
+    if ((jsonString.front() == '{') || (jsonString.front() == '//' && jsonString.back() == '}')) {
         return true;
     }
-    return testString.find(" -") != std::string::npos;
+    if (jsonString.front() == '#' || jsonString.find("\n#") != std::string_view::npos) {
+        return false;
+    }
+    auto firstColonLoc = jsonString.find_first_of(':');
+    if (firstColonLoc == std::string_view::npos) {
+        return false;
+    }
```

Comment:
A `:` doesn't need to be present in a JSON file -- if a colon isn't found, I think firstColonLoc could be set to max value for data type returned, and lastColonLoc being set to -1 (if signed numbers are supported). The ordering check at the end looks like a pretty good one for when these elements are present.